File: bwl.Imaging/Matrix/BitmapConverter.vb
Imports System.Runtime.CompilerServices

Public Class BitmapConverter
    Public Shared Function BitmapToGrayMatrix(bitmap As Bitmap) As GrayMatrix
        Dim processor As New BitmapOperations
        processor.LoadBitmap(bitmap)
        Return processor.GetGrayMatrix()
    End Function

    Public Shared Function BitmapToRGBMatrix(bitmap As Bitmap) As RGBMatrix
        Dim processor As New BitmapOperations
        processor.LoadBitmap(bitmap)
        Return processor.GetRGBMatrix()
    End Function

    Public Shared Function GrayMatrixToBitmap(matrix As GrayMatrix) As Bitmap
        Dim processor As New BitmapOperations
        processor.LoadGrayMatrixWithLimiter(matrix)
        Return processor.GetBitmap()
    End Function

    Public Shared Function RGBMatrixToBitmap(matrix As RGBMatrix, Optional useLimiter As Boolean = False) As Bitmap
        Dim processor As New BitmapOperations
        processor.LoadRGBMatrixWithLimiter(matrix)
        Return processor.GetBitmap()
    End Function
End Class

Public Module BitmapConverterExtensions
    <Extension()>
    Public Function BitmapToGrayMatrix(bitmap As Bitmap) As GrayMatrix
        Return BitmapConverter.BitmapToGrayMatrix(bitmap)
    End Function

    <Extension()>
    Public Function BitmapToRgbMatrix(bitmap As Bitmap) As RGBMatrix
        Return BitmapConverter.BitmapToRGBMatrix(bitmap)
    End Function

    <Extension()>
    Public Function ToBitmap(matrix As GrayMatrix) As Bitmap
        Return BitmapConverter.GrayMatrixToBitmap(matrix)
    End Function

    <Extension()>
    Public Function ToBitmap(matrix As RGBMatrix) As Bitmap
        Return BitmapConverter.RGBMatrixToBitmap(matrix)
    End Function
End Module

File: Bwl.Imaging/Matrix/BitmapConverter.cs
using System.Drawing;

namespace Bwl.Imaging
{

    public class BitmapConverter
    {
        public static GrayMatrix BitmapToGrayMatrix(Bitmap bitmap)
        {
            var processor = new BitmapOperations();
            processor.LoadBitmap(bitmap);
            return processor.GetGrayMatrix();
        }

        public static RGBMatrix BitmapToRGBMatrix(Bitmap bitmap)
        {
            var processor = new BitmapOperations();
            processor.LoadBitmap(bitmap);
            return processor.GetRGBMatrix();
        }

        public static Bitmap GrayMatrixToBitmap(GrayMatrix matrix)
        {
            var processor = new BitmapOperations();
            processor.LoadGrayMatrixWithLimiter(matrix);
            return processor.GetBitmap();
        }

        public static Bitmap RGBMatrixToBitmap(RGBMatrix matrix, bool useLimiter = false)
        {
            var processor = new BitmapOperations();
            processor.LoadRGBMatrixWithLimiter(matrix);
            return processor.GetBitmap();
        }
    }

    public static class BitmapConverterExtensions
    {
        public static GrayMatrix BitmapToGrayMatrix(this Bitmap bitmap)
        {
            return BitmapConverter.BitmapToGrayMatrix(bitmap);
        }

        public static RGBMatrix BitmapToRgbMatrix(this Bitmap bitmap)
        {
            return BitmapConverter.BitmapToRGBMatrix(bitmap);
        }

        public static Bitmap ToBitmap(this GrayMatrix matrix)
        {
            return BitmapConverter.GrayMatrixToBitmap(matrix);
        }

        public static Bitmap ToBitmap(this RGBMatrix matrix)
        {
            return BitmapConverter.RGBMatrixToBitmap(matrix);
        }
    }
}